From 4d0cd1c4cdfad7a166b2192ac6e05c5958c13000 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 3 Aug 2020 23:29:46 +0200
Subject: [PATCH 1/1] riscv: remove redundant logical constraint.

After

    if (ret) return ret;

we know that ret is zero. Don't check it again.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 arch/riscv/lib/andes_plic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/riscv/lib/andes_plic.c b/arch/riscv/lib/andes_plic.c
index 5cf29df670..c2a8fe4d9e 100644
--- a/arch/riscv/lib/andes_plic.c
+++ b/arch/riscv/lib/andes_plic.c
@@ -52,7 +52,7 @@ static int init_plic(void)
 	if (ret)
 		return ret;
 
-	if (ret == 0 && dev) {
+	if (dev) {
 		ofnode_for_each_subnode(node, dev_ofnode(dev->parent)) {
 			const char *device_type;
 
-- 
2.27.0

